packages/api-aco/src/record/record.so.ts

```diff
@@ -28,29 +29,41 @@ export const createSearchRecordOperations = (
         return model;
     };
 
-    const getRecord: AcoSearchRecordStorageOperations["getRecord"] = async ({ id }) => {
-        const model = await getRecordModel();
-        security.disableAuthorization();
+    const getRecord = async (id: string) => {
+        const context = getCmsContext();
+        const initialModel = await getRecordModel();
+
+        const model = attachCmsModelFieldConverters({
+            model: initialModel,
+            plugins: context.plugins
+        });
 
         /**
-         * The record "id" is generated on creation.
-         * Still, we need to get/update/delete records by the original entry id.
+         * The record "id" has been passed by the original entry.
+         * We need to retrieve it via `cms.storageOperations.entries.getLatestByIds()` method and return the first one.
          */
-        const entry = await cms.getEntry(model, { where: { originalId: id, latest: true } });
+        const revisions = await cms.storageOperations.entries.getLatestByIds(model, { ids: [id] });
 
-        if (!entry) {
-            throw new WebinyError("Could not load record.", "GET_ENTRY_ERROR", {
+        if (revisions.length === 0) {
+            throw new WebinyError("Record not found.", "NOT_FOUND", {
                 id
             });
         }
 
-        security.enableAuthorization();
-        return getFieldValues(entry, baseFields);
+        return revisions[0];
     };
 
     return {
         getRecordModel,
-        getRecord,
+        async getRecord({ id }) {
+            security.disableAuthorization();
+
+            const record = await getRecord(id);
+
+            security.enableAuthorization();
```

Comment:
I have a nice utility for you to avoid this repetitive fiddling with security: https://github.com/webiny/webiny-examples/blob/master/gated-content/packages/api/src/content/onTenantAfterInstall.ts#L6-L10